src/Microsoft.Identity.Client/Instance/B2CAuthority.cs

```diff
@@ -52,13 +52,16 @@ internal B2CAuthority(IServiceBundle serviceBundle, AuthorityInfo authorityInfo)
                 return;
             }
 
-            await base.UpdateCanonicalAuthorityAsync(requestContext).ConfigureAwait(false);
+            if (AuthorityInfo.ValidateAuthority)
```

Comment:
this is the issue. if it's not in the b2cTrustedHosts list, then it's either a custom domain or using login.microsoftonline.com. If the latter, we need to do instance discovery, if it's custom domain, we should not to instance discovery. this means the developer needs to send set validateAuthority to false for a custom domain. A dev using login.microsoftonline.com would need to pass in true (but this is being deprecated).

Reply:
How about adding withTrustedAuthority or can we default to add the authority the developer passes in to the trusted authorities list?

Reply:
How about adding withTrustedAuthority or can we default to add the authority the developer passes in to the trusted authorities list?

Reply:
Instead of introducing this extra parameter, would it not be possible to detect the host instead?

Reply:
let me think about that one. i'll get back to you.

---
In reply to: [268924586](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1012#discussion_r268924586) [](ancestors = 268924586)